packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/SelectedEntriesModal/index.js

```diff
@@ -128,26 +303,54 @@ const SelectedEntriesModalContent = ({ onToggle, onConfirm }) => {
 SelectedEntriesModalContent.propTypes = {
   onToggle: PropTypes.func.isRequired,
   onConfirm: PropTypes.func.isRequired,
+  onRefresh: PropTypes.func.isRequired,
 };
 
 /* -------------------------------------------------------------------------------------------------
  * SelectedEntriesModal
  * -----------------------------------------------------------------------------------------------*/
 
 const SelectedEntriesModal = ({ onToggle, onConfirm }) => {
-  const { rows, selectedEntries } = useTableContext();
+  const { selectedEntries } = useTableContext();
+  const { contentType } = useSelector(listViewDomain());
 
-  // Get the selected entries full data, and keep the list view order
-  // Memoize to prevent infinite useEffect runs in SelectedEntriesTableContent
-  const entries = React.useMemo(() => {
-    return rows.filter((row) => {
-      return selectedEntries.includes(row.id);
-    });
-  }, [rows, selectedEntries]);
+  // We want to keep the selected entries order same as the list view
+  const [
+    {
+      query: { sort },
+    },
+  ] = useQueryParams();
+  const queryParams = {
+    sort,
+    filters: {
+      id: {
+        $in: selectedEntries,
+      },
+    },
+  };
+
+  const { get } = useFetchClient();
+  const queryString = stringify(queryParams);
+
+  const { data, isRefetching, refetch } = useQuery(
+    ['entries', contentType.uid, queryParams],
+    async () => {
+      const { data } = await get(
+        `content-manager/collection-types/${contentType.uid}?${queryString}`
+      );
+
+      return data;
+    }
+  );
```

Comment:
I feel like this creates a UX issue. Now we don't fetch the data until this component mounts, so we have a loading state in the modal that seems unnecessary since we already had the data when the ListView component mounted.